sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -79,9 +81,16 @@ class FileStreamSource(
     sourceOptions.maxFileAgeMs
   }
 
+  private val fileNameOnly = sourceOptions.fileNameOnly
+  if (fileNameOnly) {
+    logWarning("'fileNameOnly' is enabled. Make sure your file names are unique (e.g. using " +
+      "UUID), otherwise, files using the same name will be considered as the same file and causes" +
+      " data lost")
```

Comment:
nit: if I may, this message sounds a bit odd. 
> files using the same name will be considered as the same file and causes data lost

could we say
`files with the same name but under different paths will be considered the same and causes data lost`